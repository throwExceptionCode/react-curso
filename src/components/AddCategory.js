import React, { useState } from 'react';
import PropTypes from 'prop-types';


const AddCategory = ({setCategories}) => {
    const [inputValue, setinputValue] = useState('');

    const handleInputCahnge = (e)=>{
        setinputValue(e.target.value);
    };
    const handleSubmit = (e)=>{
        e.preventDefault();
        if (inputValue.trim().length > 2 ) {
            setCategories(cats => [inputValue, ...cats] );
            setinputValue('');
        }
        
        
    };
    return (
        <form onSubmit={ handleSubmit } >
            <h2>AddCategory</h2>
                <input 
                    type='text'
                    value={ inputValue }
                    onChange={ handleInputCahnge }
                />
            </form>
    )
}

AddCategory.propTypes = {
    setCategories : PropTypes.func.isRequired,
};

export default AddCategory

