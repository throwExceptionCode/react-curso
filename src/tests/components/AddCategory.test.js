import React from 'react';
import '@testing-library/jest-dom';
import { shallow } from "enzyme";
import AddCategory from '../../components/AddCategory';



describe('Pruebas en el componente AddCategory', () => {
    const setCategories = jest.fn();
    let wrapper = shallow(<AddCategory setCategories={setCategories} />);
    
    beforeEach(()=>{
        jest.clearAllMocks();
        wrapper = shallow(<AddCategory setCategories={setCategories} />)
    });
    
    
    test('should be equal to snapshot', () => {
        expect(wrapper).toMatchSnapshot();
    }); 
    test('should be changed input text', () => {
        const input = wrapper.find('input');
        const value='hola mundo';
        input.simulate('change',{'target':{value}});
        
    });
    test('shouldnt post information', () => {
        wrapper.find('form').simulate('submit',{ preventDefault(){} });
        expect(setCategories).not.toHaveBeenCalled();
    });

    test('should call SetCategory and clear tesxt input', () => {
       const value2 = 'hola mundo2';
       wrapper.find('input').simulate('change',{'target':{value:value2}});
       wrapper.find('form').simulate('submit',{preventDefault(){}});
       expect(setCategories).toHaveBeenCalled();
       expect(setCategories).toHaveBeenCalledTimes(1);
       expect(setCategories).toHaveBeenCalledWith(expect.any(Function));
       expect(wrapper.find('input').prop('value')).toBe('');
    });
    
    
});
