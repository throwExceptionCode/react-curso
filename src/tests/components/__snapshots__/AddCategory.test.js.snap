// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente AddCategory should be equal to snapshot 1`] = `
<form
  onSubmit={[Function]}
>
  <h2>
    AddCategory
  </h2>
  <input
    onChange={[Function]}
    type="text"
    value=""
  />
</form>
`;
